qiskit/quantum_info/operators/measures.py

```diff
@@ -97,27 +97,39 @@ def process_fidelity(channel,
             "instead of `require_cptp=True`.", DeprecationWarning)
         require_cp = True
         require_tp = True
-    if isinstance(channel, Operator) and (require_cp or require_tp):
-        is_unitary = channel.is_unitary()
-        # Validate as unitary
-        if require_cp and not is_unitary:
-            raise QiskitError('channel is not completely-positive')
-        if require_tp and not is_unitary:
-            raise QiskitError('channel is not trace-preserving')
-    else:
-        # Validate as QuantumChannel
-        if require_cp and not channel.is_cp():
-            raise QiskitError('channel is not completely-positive')
-        if require_tp and not channel.is_tp():
-            raise QiskitError('channel is not trace-preserving')
+    if require_cp or require_tp:
+        # Validate target channel
+        if target is not None:
+            is_unitary = target.is_unitary()
+            if require_cp and not is_unitary:
+                raise QiskitError('Target unitary channel is not completely-positive')
+            if require_tp and not is_unitary:
+                raise QiskitError('Target unitary channel is not trace-preserving')
+        # Validate input channel
+        if isinstance(channel, Operator):
+            is_unitary = channel.is_unitary()
+            # Validate as unitary
+            if require_cp and not is_unitary:
+                raise QiskitError('Input quantum channel is not completely-positive')
+            if require_tp and not is_unitary:
+                raise QiskitError('Input quantum channel is not trace-preserving')
+        else:
+            # Validate as QuantumChannel
+            if require_cp and not channel.is_cp():
+                raise QiskitError('Input quantum channel is not completely-positive')
+            if require_tp and not channel.is_tp():
+                raise QiskitError('Input quantum channel is not trace-preserving')
 
     # Compute process fidelity with identity channel
+    if target is not None:
+        channel = channel.compose(target.adjoint())
+    input_dim, _ = channel.dim
     if isinstance(channel, Operator):
         # |Tr[U]/dim| ** 2
         fid = np.abs(np.trace(channel.data) / input_dim)**2
     else:
         # Tr[S] / (dim ** 2)
-        fid = np.trace(channel.data) / (input_dim**2)
+        fid = np.trace(SuperOp(channel).data) / (input_dim**2)
```

Comment:
Is this redundant of the `SuperOp` cast on L78?

Reply:
Yes for now, but it shouldn't cost anything. I put it there because in #5339 I originally changed the default input rep to Choi, but ended up removing that from this PR to make it safe to backport. It will end up there again when I update that PR.